.github/workflows/build.yml

```diff
@@ -373,35 +373,48 @@ jobs:
             ./coverage/lcov
             ./coverage/json
 
-  coverage:
+  codecov:
     needs: [test]
     runs-on: ubuntu-latest
     timeout-minutes: 3
     if: (success() || failure()) && github.event.pull_request.draft != true
     steps:
-      - name: Checkout code
-        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-        with:
-          sparse-checkout: ${{ env.SPARSE_CHECKOUT }}
-
       - name: Download coverage reports
         uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
         with:
           name: coverage
           path: coverage
 
       - name: Codecov
-        uses: codecov/codecov-action@e0b68c6749509c5f83f984dd99a76a1c1a231044 # v4.0.1
+        uses: codecov/codecov-action@ab904c41d6ece82784817410c45d8b8c02684457 # v3.1.6
```

Comment:
🤔 why it should work for PR's from forks without token. Do you find a issue somewhere about the broken behavior?

Reply:
- https://github.com/codecov/feedback/issues/112

Reply:
Maybe, it'll help to add the secret to Dependabot secrets, even we're not using it

Reply:
But anyway, at this point I'd prefer to fix it first, then upgrade and check